controller/userController.js

```diff
@@ -0,0 +1,85 @@
+import { userService } from '../service';
+
+const userSignUp = async (req, res) => {
+  try {
+    const {
+      email,
+      password,
+      name,
+      phoneNumber,
+      roadAddress,
+      detailAddress,
+      zipCode,
+    } = req.body;
+
+    const essentialKeys = {
+      email,
+      password,
+      name,
+      phoneNumber,
+      roadAddress,
+      detailAddress,
+      zipCode,
+    };
+
+    const values = Object.values(essentialKeys);
+    const keys = Object.keys(essentialKeys);
+
+    const keyResultOfUndefined = keys.filter((key) => {
+      return essentialKeys[key] === undefined;
+    });
+
+    if (values.includes(undefined)) {
+      let error = new Error(`KEY_ERROR ${keyResultOfUndefined}`);
+      error.statusCode = 400;
+      throw error;
+    } else {
+      await userService.userSignUp(
+        email,
+        password,
+        name,
+        phoneNumber,
+        roadAddress,
+        detailAddress,
+        zipCode
+      );
+      res.status(201).json({
+        message: 'USER_CREATED_SUCCESSFULLY',
+      });
+    }
+  } catch (error) {
+    res.status(error.statusCode || 500).json({
+      message: error.message,
+    });
+  }
+};
+
+const userLogin = async (req, res) => {
+  try {
+    const { email, password } = req.body;
+
+    if (!email && !password) {
+      let error = new Error('KEY_ERROR');
+      error.statusCode = 400;
+      throw error;
+    } else if (!email) {
+      let error = new Error('KEY_ERROR');
+      error.statusCode = 400;
+      throw error;
+    } else if (!password) {
+      let error = new Error('KEY_ERROR');
+      error.statusCode = 400;
+      throw error;
+    } else {
+      const token = await userService.userLogin(email, password);
+
+      res.status(200).json({ message: 'LOGIN_SUCCESS!', token });
+    }
```

Comment:
```suggestion
    const { email, password } = req.body;

    if (!email || !password) {
      let error = new Error('KEY_ERROR');
      error.statusCode = 400;
      throw error;
    }
    
    const token = await userService.userLogin(email, password);
    res.status(200).json({ message: 'LOGIN_SUCCESS!', token });
   
```

지현님 이 방법은 어떨까요? 둘 중에 하나라도 없으면 바로 에러를 throw하고,
그것이 아니라면 로그인을 진행하는거죠 :)

Reply:
오! 훨씬 간결하고 가독성이 좋아진 것 같습니다. 길고 길었던 에러 핸들링 함수가 굉장히 함축적으로 바뀌었네요. 왜 || 연산자를 생각하지 못했던 걸까.. 이렇게 한 번 수정해보겠습니다! 💖